emanote/src/Emanote/Model/Note.hs

```diff
@@ -147,6 +150,16 @@ queryNoteTitle r doc meta =
     withoutH1 x =
       x
 
+-- | The xml route intended by user for this note.
+noteXmlRoute :: Note -> R 'R.Xml
+noteXmlRoute note@Note {..} =
+  -- Favour slug if one exists, otherwise use the full path.
+  case noteSlug note of
+    Nothing ->
+      R.withLmlRoute coerce _noteRoute
+    Just slugs ->
+      R.mkRouteFromSlugs slugs
```

Comment:
I don't think you need this logic; just call `noteHtmlRoute` and coerce its result.